### PR TITLE
:Freat Add NN-Task

### PR DESCRIPTION
**Pull request by MrChazar**
Zabrakło mi czasu by wgłębić się w tą listę niestety :( możliwe że będę dodawać jeszcze nowe rzeczy na tygodniu. Myślałem żeby dodać inne sposoby dobierania liczby neuronów i funkcji aktywacji dla sieci

### CODE REVIEW

**Review by Hanzoze — COMMENTED** [review]
Bardzo dobrze napisana praca. Mi się spodobało że zrobiłeś analizę na liczbę neuronów i gdy trzeba przerwać czas treningu żeby nie było overfitting modelu, nawet jeżeli nie wystarczyło czasu na inne analizy, dla mnie to było nowe i ciekawe

**Review by ozio1212 — APPROVED** [review]
Tak jak w poprzedniej recenzji, też uważam że zrobiłenie analizy liczby neuronów i gdy trzeba przerwać trening żeby uniknąć overfittingu to mega fajny pomysł, który dla mnie też jest nowością. A tak poza tym to nie widać żebyś jakoś mało się przyłożył i moim zdaniem serio super praca :)

**Review by rozamazurek — COMMENTED** [review]
Fajna praca, podobało mi się dokładne sprawdzenie jakie wyniki daje model dla różnej liczby neuronów, super pomysł ze sprawdzeniem własnoręcznie napisanej liczby oraz z zastosowaniem early stoppingu. Plus za wnioski na początku i a końcu, jedyne czego mi zabrakło to może analiza wpływu jakiegoś innego parametru na sieć/model. Ale ogólnie super  praca

**Comment by GregW04** [comment]
@MrChazar  Cześć przychodzę z code review:
-  dummy +
-  podział na train-val-test + 
- przejrzysta funkcja NerualNetwork() + 
-  przez eksperyment raczej patrzy się na val a na samym końcu po budowie modeli itd. patrzysz jak się prezentują na test (ale widzę ze pozniej już dobrze robisz 👍)
-  własnoręczna 5 (bardzo fajny pomysł, podejście do danych zawsze powinno być lekko sceptyczne) +
- metryki (Classification report) +
- macierz pomyłek +
- EarlyStopping + 
- Szacun za te eksperymenty z liczbą neuronów (powiedzmy ze szerokosc tym zbadana) + 
- brak badania głębokości sieci - 
-  brak badania z użyciem krzywych uczenia (metryka+ loss na train i val) - 
- fajnie ze pokazałeś które liczby są najczęściej mylone i dlaczego +
- celne i dobre komentarze, przejrzysty kod +
- Brak BatchNormalization -
- Brak Dropout -
- Co bym dodał:
- tam gdzie masz minusy
- pobadał różne optimizery i ich learning-rate'y
-  wykresy słupkowe(metryka i loss) dla znacznych epok (1 i ostatnia) dla różnych opcji modeli
- eksperymenty z różnymi funkcjami aktywacji
- ogólnie cała praca czytelna z naciskiem na szczegóły (w tym przypadku fajnie ze przyjrzałeś się głębiej  problemowi) + 
- Widzę ze zabrakło ci czasu aby zrobić więcej, jeżeli chcesz to zrób więcej i mnie oznacz po dodatkowe review
Dzięki
